src/page/product-list.page.ts

```diff
@@ -1,13 +1,28 @@
 import { $, ElementFinder } from 'protractor';
 
 export class ProductListPage {
-  private shirt: ElementFinder;
+  private products: ElementFinder;
+  private addButton: ElementFinder;
 
   constructor () {
-    this.shirt = $('#center_column ul.product_list > li:nth-child(1) a.button.ajax_add_to_cart_button.btn.btn-default');
+    this.products = $('#center_column ul.product_list');
+    this.addButton = $('#add_to_cart > button');
   }
 
-  public async goToShirt(): Promise<void> {
-    await this.shirt.click();
+  public async findByProduct(name:string) {
+    return await this.products.$$('.product-container').filter(async (product) => {
+      return await product.$('.product-name').getText().then((productName) => {
+        return productName === name;
+      });
+    }).first();
+  }
+
+  public async selectProduct(name:string) {
+    const product = this.findByProduct(name);
+    await (await product).$('.product-image-container .replace-2x').click();
```

Comment:
Why do you have a double await?

Reply:
(await product) is for the product initialization, the same as doing const product = await this.find...